https://github.com/biomejs/biome/issues/6465 The main export [imports](https://github.com/biomejs/biome/blob/%40biomejs/js-api%406.0.0/packages/%40biomejs/js-api/src/index.ts#L1-L12) the types for the three dependencies: `"@biomejs/wasm-bundler"`, `"@biomejs/wasm-nodejs"`, and `"@biomejs/wasm-web"`.

In my project, I only have the dependencies [`"@biomejs/js-api"` and `"@biomejs/wasm-nodejs"`](https://github.com/regseb/metalint/blob/v0.21.2/package.json#L70-L71). And I'm using the subpath [`"@biomejs/js-api/nodejs"`](https://github.com/regseb/metalint/blob/v0.21.2/src/core/wrapper/biomejs__js-api.js#L12). So if I use the main export to get the `spanInBytesToSpanInCodeUnits` function, TypeScript will throw an error: _Cannot find module '@biomejs/wasm-web'_

I would like to:

```javascript
import { Biome, spanInBytesToSpanInCodeUnits } from "@biomejs/js-api/nodejs";
```

---

To reproduce the problem:

- `package.json`

  ```json
  {
    "name": "testcase",
    "version": "1.0.0",
    "type": "module",
    "dependencies": {
      "@biomejs/js-api": "6.0.0",
      "@biomejs/wasm-nodejs": "2.5.0",
      "typescript": "6.0.3"
    }
  }
  ```

- `index.ts`

  ```typescript
  import { Biome } from "@biomejs/js-api/nodejs";
  import { spanInBytesToSpanInCodeUnits } from "@biomejs/js-api";
  ```

1. `npm install`
2. `npx tsc -lib esnext index.ts`

   ```
   node_modules/@biomejs/js-api/dist/index.d.ts:1:93 - error TS2307: Cannot find module '@biomejs/wasm-bundler' or its corresponding type declarations.

   1 import type { Configuration as ConfigurationBundler, Diagnostic as DiagnosticBundler } from "@biomejs/wasm-bundler";
                                                                                                 ~~~~~~~~~~~~~~~~~~~~~~~

   node_modules/@biomejs/js-api/dist/index.d.ts:3:85 - error TS2307: Cannot find module '@biomejs/wasm-web' or its corresponding type declarations.

   3 import type { Configuration as ConfigurationWeb, Diagnostic as DiagnosticWeb } from "@biomejs/wasm-web";
                                                                                         ~~~~~~~~~~~~~~~~~~~


   Found 2 errors in the same file, starting at: node_modules/@biomejs/js-api/dist/index.d.ts:1
   ```

--- a/packages/@biomejs/js-api/README.md
+++ b/packages/@biomejs/js-api/README.md
@@ -22,8 +22,8 @@ You need to install one of the `@biomejs/wasm-*` package as a **peer dependency*
 ```js
 import { Biome } from "@biomejs/js-api/nodejs";
 // Or:
-// import { Biome, Distribution } from "@biomejs/js-api/bundler";
-// import { Biome, Distribution } from "@biomejs/js-api/web";
+// import { Biome } from "@biomejs/js-api/bundler";
+// import { Biome } from "@biomejs/js-api/web";
 
 const biome = new Biome();
 const { projectKey } = biome.openProject("path/to/project/dir");
@@ -51,6 +51,21 @@ const html = biome.printDiagnostics(result.diagnostics, {
 });
 
 console.log("Lint diagnostics: ", html);
+```
+
+If you want to work with diagnostics in your program, you can use the `spanInBytesToSpanInCodeUnits` helper function to convert byte-based segments from Biome diagnostics into UTF-16 code unit segments (used in JavaScript).
+
+```js
+import { Biome, spanInBytesToSpanInCodeUnits } from "@biomejs/js-api/nodejs";
+
+for (const diagnostic of result.diagnostics) {
+  const [start, end] = spanInBytesToSpanInCodeUnits(
+    diagnostic.location.span,
+    formatted.content,
+  );
+  // Correctly extracts the text
+  const text = formatted.content.slice(start, end);
+}
 ```
 
 ## Philosophy

